packages/nextjs/app/useCofhe.ts

```diff
@@ -0,0 +1,330 @@
+"use client";
+
+import { useCallback, useEffect, useMemo, useSyncExternalStore } from "react";
+import { Result } from "@cofhe/sdk";
+import { hardhat } from "@cofhe/sdk/chains";
+import {
+  CreateSelfPermitOptions,
+  CreateSharingPermitOptions,
+  Permit,
+  PermitUtils,
+  permitStore,
+} from "@cofhe/sdk/permits";
+import { createCofhesdkClient, createCofhesdkConfig } from "@cofhe/sdk/web";
+import { WalletClient, createWalletClient, http } from "viem";
+import { PrivateKeyAccount, privateKeyToAccount } from "viem/accounts";
+import * as chains from "viem/chains";
+import { useAccount, usePublicClient, useWalletClient } from "wagmi";
+import { create } from "zustand";
+import scaffoldConfig from "~~/scaffold.config";
+import { logBlockMessage, logBlockMessageAndEnd, logBlockStart } from "~~/utils/cofhe/logging";
+import { notification } from "~~/utils/scaffold-eth";
+
+const zkvSignerPrivateKey = "0x6C8D7F768A6BB4AAFE85E8A2F5A9680355239C7E14646ED62B044E39DE154512";
+function createWalletClientFromPrivateKey(privateKey: `0x${string}`): WalletClient {
+  const account: PrivateKeyAccount = privateKeyToAccount(privateKey);
+  const url = chains.hardhat.rpcUrls.default.http[0];
+  return createWalletClient({
+    account,
+    chain: chains.hardhat,
+    transport: http(url),
+  });
+}
+const mockViemZkvSigner = createWalletClientFromPrivateKey(zkvSignerPrivateKey);
```

Comment:
We can remove all this zkvSigner stuff. This is handled internally in the @cofhe/sdk/web createCofhesdkClient so it doesn't need to be manually done here.

Reply:
Shouldn't need to import `hardhat-plugin`. Probably better to put the zkv signer private key in the sdk and export it. Then it can be imported by `hardhat-plugin` instead of being hard coded.

I do think the right place for constructing the mock zkvSigner is in node/web createCofheConfig. It should be overridable if needed (hardhat plugin overrides it to use the hardhat ethers signe), but I don't think that will be a common pattern for the client devs to do.

Reply:
ok, this is almost [what we ended up with](https://github.com/FhenixProtocol/cofhesdk/pull/35) 👌